### PR TITLE
Bugfix

### PR DESCRIPTION
**Pull request by fulmicoton**
- Update the control plane model on toggle source.
- Delete source from index config in control plane model on delete source.

### CODE REVIEW

File: quickwit/quickwit-control-plane/src/control_plane.rs [hunk 1]
```diff
@@ -317,14 +317,22 @@ impl Handler<ToggleSourceRequest> for ControlPlane {
...
         };
 
+        let has_changed = self.model.toggle_source(&index_uid, &source_id, enable)?;
+
         // TODO update the internal view.
         // TODO: Refine the event. Notify index will have the effect to reload the entire state from
         // the metastore. We should update the state of the control plane.
-        self.indexing_scheduler.on_index_change(&self.model).await?;
+        if has_changed {
+            self.indexing_scheduler.on_index_change(&self.model).await?;
```
Comment (guilload) [inline quickwit/quickwit-control-plane/src/control_plane.rs:334]: Do we want to return an error if the `toggle_source` call has succeeded but the scheduling failed? I have left a todo for the same thing below.

File: quickwit/quickwit-control-plane/src/control_plane_model.rs [hunk 2]
```diff
@@ -245,7 +246,34 @@ impl ControlPlaneModel {
...
+    /// Returns `true` if the source status has changed, `false` otherwise.
+    /// Returns an error if the source could not be found.
+    pub(crate) fn toggle_source(
+        &mut self,
+        index_uid: &IndexUid,
+        source_id: &SourceId,
+        enable: bool,
+    ) -> anyhow::Result<bool> {
+        let Some(index_model) = self.index_table.get_mut(index_uid) else {
+            bail!("Index `{}` not found.", index_uid);
```
Comment (guilload) [inline quickwit/quickwit-control-plane/src/control_plane_model.rs:270]:
```suggestion
            bail!("index `{index_uid}` not found");
```

File: quickwit/quickwit-control-plane/src/control_plane_model.rs [hunk 3]
```diff
@@ -245,7 +246,34 @@ impl ControlPlaneModel {
...
+        };
+        let Some(source_config) = index_model.sources.get_mut(source_id) else {
+            bail!("Source `{}` not found.", source_id);
```
Comment (guilload) [inline quickwit/quickwit-control-plane/src/control_plane_model.rs:273]:
```suggestion
            bail!("source `{source_id}` not found.");
```

File: quickwit/quickwit-control-plane/src/control_plane_model.rs [hunk 4]
```diff
@@ -245,7 +246,34 @@ impl ControlPlaneModel {
...
+        };
+        source_config.enabled = enable;
+        Ok(enable)
```
Comment (guilload) [inline quickwit/quickwit-control-plane/src/control_plane_model.rs:276]: This does not return whether the source has changed but rather whether it was enabled or not.

**Review by guilload — APPROVED** [review]

File: quickwit/quickwit-control-plane/src/control_plane_model.rs (hunk 4 above)
Comment (fulmicoton) [reply quickwit/quickwit-control-plane/src/control_plane_model.rs:276]: that's a bug indeed.

File: quickwit/quickwit-control-plane/src/control_plane.rs (hunk 1 above)
Comment (fulmicoton) [reply quickwit/quickwit-control-plane/src/control_plane.rs:331]: Good point. 
Actually this does not need to be async and does not return an error.
Addressing htat in a different PR.

**Pull request merged**